Mathlib/LinearAlgebra/Projection.lean

```diff
@@ -370,10 +370,10 @@ open Submodule
 /--
 A linear endomorphism of a module `E` is a projection onto a submodule `p` if it sends every element
 of `E` to `p` and fixes every element of `p`.
-The definition allow more generally any `FunLike` type and not just linear maps, so that it can be
+The definition allow more generally any `DFunLike` type and not just linear maps, so that it can be
 used for example with `ContinuousLinearMap` or `Matrix`.
 -/
-structure IsProj {F : Type*} [FunLike F M fun _ => M] (f : F) : Prop where
+structure IsProj {F : Type*} [DFunLike F M fun _ => M] (f : F) : Prop where
```

Comment:
Should this be dependent? Might be out of scope for this PR

Reply:
I'd like to postpone this cleanup after the heavy changes of #8386 have passed.